be/src/olap/options.cpp

```diff
@@ -17,26 +17,35 @@
 
 #include "olap/options.h"
 
-#include <boost/algorithm/string.hpp>
-#include <boost/algorithm/string/classification.hpp>
-#include <boost/algorithm/string/split.hpp>
+#include <algorithm>
 
 #include "common/logging.h"
-
 #include "olap/utils.h"
 
+#include "gutil/strings/split.h"
+
 namespace doris {
 
+static std::string CAPACITY_UC = "CAPACITY";
+static std::string MEDIUM_UC = "MEDIUM";
+static std::string SSD_UC = "SSD";
+static std::string HDD_UC = "HDD";
+
+// TODO: should be a general util method
+std::string to_upper(const std::string& str) {
+    std::string out = str;
+    std::transform(out.begin(), out.end(), out.begin(), ::toupper);
```

Comment:
why not use `for ()`. you copy the string two times.

Reply:
It's only copied once. It's modified in place in `transform()`

Reply:
I know, the first copy could be avoided if the transformed character is written to the destination string when iterate the input string.
So, I think it is not a good idea to use `transform` here, because the input string is iterated twice.

Reply:
I am not sure if I get what you mean, Is it the following way?
```
std::string out;
out.resize(str.size());
for (int i = 0; i < str.size(); ++i) {
    out[i] = ::toupper(str[i]);
}
return out;
```

I am not to be overly pedantic, but it still will iterate twice.

The first one is in `std::string::resize()`, all the character will be set to '\0'(i.e. it need a `memset()`), like the Ctor `std::string(const string& str)`(it need a `memcpy()`)

And, the above implementation may have a side effect: it may dynamically apply for memory twice.
- The first time in the std::string's `default ctor`, apply for a default capacity;
- The second time is in `resize()` if the required length is greater than the default capacity

Reply:
Oh yes, you are right, I forget the memset operation of operation.
However, if replace `out.resize(str.size());` with `out.reserve(str.size());` in your case. This function will only iterate once.

Reply:
Emm, if using `reserve`, something trick should be used. So I think your code is preferred.